### DISCUSSION TITLE
Run UERANSIM without sudo

### DISCUSSION THREAD

**Discussion by GiovanniColonni** [aligungr/UERANSIM#488] [Q&A] [answered]
I'm currently try to run _nr-ue_ without using sudo but I'm facing this problem:

`[error] TUN interface could not be setup. Permission denied. Please run the UE with 'sudo'`

I need this in order to run the UEs automatically from a script.
The fact is that I don't get what is the file that I have to modify and in which way.

My system:
Ubuntu 21.10, impish

(btw thanks for this amazing project, I love it :heart_eyes:)

**Comment by aligungr** [comment, marked as answer]
Hello @GiovanniColonni 

For security reasons, it's not possible to create a TUN interface with the standard privileges in Linux systems.

BR

**Reply by GiovanniColonni** [reply]
Ok thank you, do you have ideas on how to automate the task of running gNodeB and UEs from, for example, an ssh command?

**Reply by GiovanniColonni** [reply]
I found [this](https://askubuntu.com/questions/452345/how-to-automate-sudo-password), in this way you still have to use sudo but without inserting the password.
I leave it as a reference.
